packages/firestore/src/local/persistence_promise.ts

```diff
@@ -238,4 +238,28 @@ export class PersistencePromise<T> {
     }
     return p;
   }
+
+  /**
+   * Given a collection that has a `forEach` method, call the given function on
+   * each element in the collection and wait for all of the resulting
+   * concurrent PersistencePromises to resolve.
+   */
+  static waitForEach<K>(
```

Comment:
I just revived one of my older PRs that would allow you to merge this with `forEach` above. Let me know if you think this could work for you: https://github.com/firebase/firebase-js-sdk/pull/1269

Reply:
I feel like my comment on this got lost somehow. If you do think that the parallelism improves things, can we make the difference between `forEach`/`waitFor`/`waitForEach` more explicit? We could maybe rename `waitForEach` to `forEachParalellel` or something similar (sorry for the horrible name suggestion) or pass in a flag that allows parallelization. For what it's worth, there are currently other calls to `forEach` that could be parallelized.